Flang runtime depends on libomp but does not link it with `-rpath`

`libflang.so` and `libflangrti.so` contain references to `libomp.so` symbols but, when the driver is invoked without `-fopenmp`, the linker does not attempt to link a user program with the OpenMP runtime. In the absence of `LD_RUN_PATH` or `LD_LIBRARY_PATH`, building any user program without `-fopenmp` results in linker errors like the ones below:
```
/usr/bin/ld: warning: libomp.so, needed by /home2/b00375952/src/install.release_13x/bin/../lib/libflang.so, not found (try using -rpath or -rpath-link)
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_set_dynamic@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_set_num_threads@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_get_schedule@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_set_max_active_levels@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_get_ancestor_thread_num@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_set_nested@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_get_team_size@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_threadprivate@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `omp_init_nest_lock@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_barrier@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `omp_unset_nest_lock@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_end_critical@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_critical@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_global_thread_num@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `__kmpc_threadprivate_cached@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflang.so: undefined reference to `omp_set_schedule@VERSION'
/home2/b00375952/src/install.release_13x/bin/../lib/libflangrti.so: undefined reference to `omp_set_nest_lock@VERSION'
```

--- a/runtime/flang/CMakeLists.txt
+++ b/runtime/flang/CMakeLists.txt
@@ -721,3 +721,4 @@ target_compile_options(flang_static PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:-Mreent
 
 target_compile_options(flang_shared PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:-Mreentrant>)
 
+set_target_properties(flang_shared PROPERTIES INSTALL_RPATH "$ORIGIN")

--- a/runtime/flangrti/CMakeLists.txt
+++ b/runtime/flangrti/CMakeLists.txt
@@ -155,3 +155,4 @@ target_compile_options(flangrti_static PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:-Mre
 
 target_compile_options(flangrti_shared PUBLIC $<$<COMPILE_LANGUAGE:Fortran>:-Mreentrant>)
 
+set_target_properties(flangrti_shared PROPERTIES INSTALL_RPATH "$ORIGIN")

